CommonMark/Formatters/HtmlFormatterSlim.cs

```diff
@@ -651,6 +659,33 @@ private static void InlinesToHtml(HtmlTextWriter writer, Inline inline, CommonMa
             }
         }
 
+        private static void WriteGitHubHeadingIds(HtmlTextWriter writer, Inline inline, CommonMarkSettings settings, Stack<InlineStackEntry> stack) {
+            using (var tempWriter = new System.IO.StringWriter()) {
+                var tempWrapper = new HtmlTextWriter(tempWriter);
+                var tempStringBuilder = new StringBuilder();
+                InlinesToPlainText(tempWrapper, inline, stack);
+                string plaintextContent = tempWriter.ToString();
+
+                // Normalize plaintext content according to GitHub ID rules
+                plaintextContent = new Regex(@"[^\w\-\ ]").Replace(plaintextContent, "").Replace(" ", "-");
```

Comment:
you are already using a stringbuilder (though not specifying initial capacity...)